src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -366,6 +369,11 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("hello there|hello again|hello|hello|hello|hello", "hello(?> there| again|)")]
         [InlineData("hello there|hello again|hello|hello|hello|hello|hello world", "hello(?> there| again|| world)")]
         [InlineData("hello there|hello again|hello|hello|hello|hello|hello world|hello", "hello(?> there| again|| world)")]
+        [InlineData("ab|cd|||ef", "ab|cd||ef")]
+        [InlineData("|ab|cd|e||f", "|ab|cd|ef")]
```

Comment:
empty at the end as well? 
```suggestion
        [InlineData("|ab|cd|e||f|", "|ab|cd|ef")]
```

Reply:
eh, it's covered.